controllers/reconcile_fake_test.go

```diff
@@ -0,0 +1,286 @@
+package controllers_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	tlog "github.com/go-logr/logr/testing"
+	"github.com/google/go-cmp/cmp"
+	konsumeratorv1alpha1 "github.com/lwolf/konsumerator/api/v1alpha1"
+	"github.com/lwolf/konsumerator/controllers"
+	"github.com/lwolf/konsumerator/pkg/helpers"
+	"github.com/lwolf/konsumerator/pkg/helpers/tests"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func initReconciler(consumer *konsumeratorv1alpha1.Consumer) (client.Client, *controllers.ConsumerReconciler) {
+	var s = runtime.NewScheme()
+
+	_ = clientgoscheme.AddToScheme(s)
+	_ = konsumeratorv1alpha1.AddToScheme(s)
+	_ = appsv1.AddToScheme(s)
+
+	objs := []runtime.Object{
+		consumer,
+	}
+	cl := fake.NewFakeClientWithScheme(s, objs...)
+	broadcaster := record.NewBroadcasterForTests(time.Second)
+	eventSource := corev1.EventSource{Component: "eventTest"}
+	recorder := broadcaster.NewRecorder(s, eventSource)
+	return cl, &controllers.ConsumerReconciler{cl, tlog.NullLogger{}, recorder, s}
+
+}
+
+func TestConsumerReconciliation(t *testing.T) {
+	var (
+		name      = "testconsumer"
+		namespace = "testns"
+	)
+	testCases := map[string]struct {
+		consumer              *konsumeratorv1alpha1.Consumer
+		expDeployAnnotation   map[string]map[string]string
+		expContainerResources map[string]corev1.ResourceRequirements
+		expContainerEnv       map[string]map[string]string
+	}{
+		"deployment should have minimum resources without metrics provider": {
+			consumer: &konsumeratorv1alpha1.Consumer{
```

Comment:
looks like `ObjectMeta` and `Spec` are equal for all tests - let's move them to default struct